components/Donate.vue

```diff
@@ -64,83 +76,99 @@ watch(
       Array<`0x${string}`>,
       Array<`0x${string}`>
     ],
-  ([chainId, accounts, contextAccounts]: [
+  async ([chainId, accounts, contextAccounts]: [
     number,
     Array<`0x${string}`>,
     Array<`0x${string}`>
   ]) => {
     walletConnected.value =
       accounts?.length > 0 &&
       contextAccounts?.length > 0 &&
-      (chainId === 42 || chainId === 4201);
-  }
-);
+      (chainId === 42 || chainId === 4201)
 
-const error = ref(""); // Error message for validation feedback
-const amount = ref(1);
+    const { profile: _profile }: ProfileQuery = await GqlProfile({
```

Comment:
You probably want to debounce this call. The watch callback will happen multiple times if things like accounts, contextAccounts, chainId change.
```js
<script setup>
import { ref, watch } from 'vue';

// Debounce function
function debounce(fn, delay) {
  let timeout;
  return (...args) => {
    clearTimeout(timeout);
    timeout = setTimeout(() => {
      fn(...args);
    }, delay);
  };
}

const input = ref('');
const debouncedInput = ref('');

// Watch with debounce
watch(
  input,
  debounce((newValue) => {
    debouncedInput.value = newValue;
    console.log('Debounced value:', newValue);
  }, 300)
);
</script>

<template>
  <div>
    <input v-model="input" placeholder="Type here" />
    <p>Debounced Input: {{ debouncedInput }}</p>
  </div>
</template>
```
something like this maybe?
```suggestion
    const { profile: _profile }: ProfileQuery = await GqlProfile({
```

Reply:
It's ok when just setting a flag, but making multiple queries is harder. Another way to do it would be to have an independent watch for the profile address only, which would not change unless the address changed.

Reply:
I would actually move this network switch into composable, we will probably need it in other apps. But we can refactor later too.